src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -652,15 +659,23 @@ std::unique_ptr<model_spec> activity_classifier::init_model() const
   size_t prediction_window = read_state<flex_int>("prediction_window");
   const flex_list &features_list = read_state<flex_list>("features");
 
-  // Initialize a random number generator for weight initialization.
-  std::seed_seq seed_seq = { read_state<int>("random_seed") };
-  std::mt19937 random_engine(seed_seq);
-
+  //Only to create a random engine for weight initialization if use_zero_init = false
+  std::mt19937 random_engine;
+  if (use_zero_init == false){
```

Comment:
Our code tends to the style `if (!use_zero_init) {`

Might be more clear to use the inverse: `use_random_init`